File: pec-observation-app/services/ms-002-school/app/models/teacher.py
from datetime import datetime
from sqlalchemy import Boolean, Column, DateTime, JSON, String
from pec_shared.models_base import Base, gen_uuid


class Teacher(Base):
    __tablename__ = "teachers"

    id = Column(String, primary_key=True, default=gen_uuid)
    school_id = Column(String, nullable=False, index=True)
    name = Column(String, nullable=False)
    registration_number = Column(String, nullable=True)  # RF number
    subjects = Column(JSON, default=list)
    grades = Column(JSON, default=list)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: pec-observation-app/ios-app/Sources/PECObservation/Domain/Entities/Teacher.swift
import Foundation

public struct Teacher: Identifiable, Equatable, Codable {
    public let id: UUID
    public let schoolId: UUID
    public var name: String
    public var registrationNumber: String?
    public var subjects: [String]
    public var grades: [String]

    public init(id: UUID = UUID(), schoolId: UUID, name: String,
                registrationNumber: String? = nil,
                subjects: [String] = [], grades: [String] = []) {
        self.id = id
        self.schoolId = schoolId
        self.name = name
        self.registrationNumber = registrationNumber
        self.subjects = subjects
        self.grades = grades
    }
}
